GNUmakefile

```
@@ -74,21 +73,22 @@ go-solidity-wrappers-ocr2vrf: abigen ## Recompiles solidity contracts and their
 testdb: ## Prepares the test database.
 	go run ./core/main.go local db preparetest
 
-.PHONY: testdb
+.PHONY: testdb-user-only
 testdb-user-only: ## Prepares the test database with user only.
 	go run ./core/main.go local db preparetest --user-only
 
 # Format for CI
 .PHONY: presubmit
-presubmit: ## Format go files and imports.
-	goimports -w ./core
-	gofmt -w ./core
-	go mod tidy
+presubmit: gomodtidy ## Format go files and imports.
+	goimports -w -local github.com/smartcontractkit/chainlink .
```

What about generating _after_ running goimports, so that any incidental reformatting of generated files is undone?
```suggestion
	goimports -w -local github.com/smartcontractkit/chainlink .
	make generate
```

---

Seems hacky and bad UX if it slows things down significantly though.... :thinking:

---

I added a "make presubmit" after each `go generate` in the makefile.  Also added a `make generate` command for running `go generate ./...` on the whole repo.  As long as we always run it like that, should be consistent.

I also did a total rework of the gethwrappers2 directory, based on suggestions from @connorwstein .  Since these files are already generated in `libocr`, it just copies them from there now instead of compiling them independently.

There is one "legacy" wrapper that isn't in libocr (`offchainaggregator.go`), but we still use for some tests.  This is deprecated, so hopefully we'll eventually get rid of it but for now it's just preserved as-is, and won't ever get regenerated.

Also:  commit list was getting long, squashed everything into one commit, and cleaned up the comments

---

> As long as we always run it like that, should be consistent.

I'm afraid it may not be that easy. We are fighting against the tooling by going this route. Generated files are not supposed to be edited, and individual generate statements or sets of statements are supposed to be executable independently.

---

I'm still struggling to understand why this is +60,000 lines. IMHO these changes need to be separated to be reviewed properly.

---

Wow, didn't realize it was that huge, but yeah... but yeah, I think it will be much easier to review if I separate out the manually changed things in one commit and have the generated stuff in a second commit.  How I would suggest reviewing it is, just look at the manual commit, then run the same commands locally to generate the second commit and make sure there is no diff.   Does that sound good?  Or do you mean break it up into multiple PR's?

---

As for fighting with the tooling, I think that's true no matter what we do here since we have two tools, `go generate` and `goimports`, and they each want to fight with each other... there is no way to please them both.  `goimports` seems like a better choice to me to please.  It runs much faster than `go generate`, so the overhead is minor to just automatically add as a finishing touchup for when `go generate` is run.  It would be unreasonable to run `go generate` on the whole repo every time we run `goimports`.  Especially since going forward, we'd want the standard practice to be:   always run `make presubmit` before submitting each commit, so it gets formatted.  We can't run `go generate` every commit, it's way too slow and includes compiling solidity contracts among other things.

Perhaps there's some problem with reformatting mock files I'm unaware of, but it seems fine to me... is there a particular tool you're thinking might not like that?

---

I guess the two possibilities for why it's 60,000 lines are:  either nobody has run goimports before, or I'm doing something differently.  If it's the latter I don't know what it would be... it would be good to have someone else run it locally to make sure it's not dependent on environment.

---

Separate commits is fine. I think we need to see things that way to understand what is happening in the first place.

> for fighting with the tooling, I think that's true no matter what we do here since we have two tools, go generate and goimports, and they each want to fight with each other... there is no way to please them both. goimports seems like a better choice to me to please.

We are optionally over constraining goimports and the linter. We can already ignore mock and generated from the linter, so I think we should seriously consider just dropping the prefix sort from goimports

---

> Perhaps there's some problem with reformatting mock files I'm unaware of, but it seems fine to me... is there a particular tool you're thinking might not like that?

Everything? The generated files all contain loud headers about "do not edit", dev and CI tools assume they can regenerate them and get clean state, etc.

---

> I guess the two possibilities for why it's 60,000 lines are:  either nobody has run goimports before, or I'm doing something differently.  If it's the latter I don't know what it would be... it would be good to have someone else run it locally to make sure it's not dependent on environment.

This is why we need the changes separate. The only thing that's makes sense to me is if abigen is generating new API that we don't use in more recent versions. Otherwise it looks like a bunch of extra noise coming in.

---

> Especially since going forward, we'd want the standard practice to be: always run make presubmit before submitting each commit, so it gets formatted.

I understand this would be convenient, but is it really practical or necessary compared to our current flow? AFAIK most developers have their IDE formatting on save, and never have to think about this. Plus the per commit goal is still only to format the files you are changing. I.e. when the formats change, wouldn't we rather make a deliberate repo wide PR with intention, instead of the next random PR reformatting everything incidentally?

---

> The only thing that's makes sense to me is if abigen is generating new API that we don't use in more recent versions.

Yeah, this is definitely the case.  Almost none of the contracts even have the same name any more.  The ones we had were just super old, and unused... they were from very early stage development of OCR2 but have been replaced with new contracts with more elaborate API's.

---

The fact that they got so out of date, was one reason I felt like it was important to make it easier to keep things up to date.  My plan now is to just get rid of gethwrappers2 since it's a duplicate of what's in libocr.  That will get rid of over 60K lines.  I think the reason it was there originally was that OCR2 wasn't a part of any public repo yet. 
 Now that it's public, we should be able to import the contract wrappers directly from there.  (Although come to think of it, I'm not sure why we don't do that for OCR1, hmmm...)